src/Tribe/Body_Classes.php

```diff
@@ -0,0 +1,223 @@
+<?php
+/**
+ * Class used to manage and body classes across our plugins.
+ *
+ * @since TBD
+ */
+namespace Tribe;
+
+if ( ! class_exists( 'Body_Classes' ) ) {
+	class Body_Classes {
+		/**
+		 * Stores all the classes.
+		 * In the format: ['class' => true, 'class => false ]
+		 *
+		 * @var array
+		 */
+		protected $classes = [];
+
+		/**
+		 * Static singleton factory method.
+		 *
+		 * @since TBD
+		 *
+		 * @return self
+		 */
+		public static function instance() {
```

Comment:
Do not create `instance` methods.
This class is instantiated as a singleton only to make sure it does not hook twice on `body_class`; this whole method can be avoided using the container/service-locator.
See below.